index.js

```diff
@@ -30,7 +31,12 @@ Gulp.prototype.watch = function(glob, opt, task) {
     fn = this.parallel(task);
   }
 
-  return vfs.watch(glob, opt, fn);
+  var watcher = chokidar.watch(glob, opt);
+  if (fn) {
+    watcher.on('all', fn);
```

Comment:
I thought a little about it. The most basic pipeline is copying from A to B, and this includes files *and* folders, even if they're empty. So if you can pipe empty folders, you might also want to watch adding new directories in case you want to sync them on folder level.

However, this is also a rather rare use case. At the moment, the PR features @TrySound's suggestion.